tests/unit/src/test/scala/tests/RenameSuite.scala

```diff
@@ -0,0 +1,519 @@
+package tests
+import scala.concurrent.Future
+
+object RenameSuite extends BaseLspSuite("rename") {
+
+  renamed(
+    "basic",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |object Main{
+       |  val <<toRename>> = 123
+       |}
+       |/a/src/main/scala/a/Main2.scala
+       |package a
+       |object Main2{
+       |  val toRename = Main.<<toR@@ename>>
+       |}
+       |""".stripMargin,
+    newName = "otherRename"
+  )
+
+  renamed(
+    "across-targets",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |object Main{
+       |  val <<toRename>> = 123
+       |}
+       |/b/src/main/scala/b/Main2.scala
+       |package b
+       |import a.Main
+       |object Main2{
+       |  val toRename = Main.<<toR@@ename>>
+       |}
+       |""".stripMargin,
+    newName = "otherRename"
+  )
+
+  renamed(
+    "unapply",
+    """|/a/src/main/scala/a/Main.scala
+       |object <<F@@oo>> {
+       |  def unapply(s: String): Option[String] = Some("")
+       |}
+       |
+       |object Main{
+       |  "foo" match {
+       |    case <<Foo>>(s) => ()
+       |  }
+       |}
+       |""".stripMargin,
+    newName = "Bar"
+  )
+
```

Comment:
Missing test case: named parameter for case class fields

```scala
case class Name(value: String)
Name(value = "42")
  .copy(value = "43")
  .value
new Name(value = "44")
```

Reply:
It seems that references are not being found for 42 nad 44. Added that test case.

Reply:
Seems to be the same issue as https://github.com/scalameta/scalameta/issues/1169